src/main/java/fi/helsinki/cs/tmc/cli/backend/TmcUtil.java

```diff
@@ -216,34 +233,46 @@ private static void handleTmcExceptions(CliContext ctx, Exception exception) {
             io.errorln("Your username or password is not valid anymore.");
             return;
         }
+
+        if (exception instanceof IllegalArgumentException) {
+            logger.error("Invalid arguments", exception);
+            io.errorln("Please give server, username and password in valid form.");
+            return;
+        }
+
         if (cause instanceof FailedHttpResponseException) {
             logger.error("Unable to connect to server", exception);
             io.errorln("Unable to connect to server.");
             return;
         }
+
+        if (exception instanceof UnknownHostException) {
+            logger.error("Unknown host", exception);
+            io.errorln("Unknwon host, check the server address.");
+            return;
+        }
+
         if (cause instanceof ObsoleteClientException) {
             logger.error("Outdated tmc client");
             io.errorln("Your tmc-cli is outdated. Please update it.");
             ctx.getApp().runAutoUpdate();
             return;
         }
+
         if (cause != null && cause.getCause() instanceof UnknownHostException) {
             logger.error("No internet connection");
             io.errorln("You have no internet connection.");
             return;
         }
         logger.error("Command failed in tmc-core", exception);
         //TODO we seem to write twice error message; here and in the commands.
-        io.errorln("Command failed in tmc-core, check tmc-cli.log file for more info");
+        io.errorln("Command failed, check tmc-cli.log file for more info");
     }
 
     private static boolean isAuthenticationError(Exception exception) {
```

Comment:
Could be simplified to `return exception.getCause() instanceof OAuthProblemException`.